sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,415 @@
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Overview
+
+This extension defines free functions for load/store operations within work
+group scope.
+
+== Single Value Functions
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API scalar
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = decltype(properties())>
+void group_load(Group g, InputIteratorT in_ptr,
+                      OutputT& out, Properties = {}); // (1)
+
+// Store API scalar
+template<typename Group, typename InputT,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const InputT& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (2)
+
+}
+----
+
+(1) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+._Constraints_:
+* Only available if `Group` is a work-group or sub-group.
+* `InputIteratorT` suppose to be a random access iterator.
+* Value type of `InputIteratorT` must be convertible to OutputT.
+
+_Effects_: Loads single element from `in_ptr` to `out` by using the `g` group
+object to identify memory location as `in_ptr` + `g.get_local_linear_id()`.
+
+`Properties` argument is reserved for future revisions of this extention and is
+ignored now. Default value is empty `sycl::ext::oneapi::experimental::properties<>`
+May be used in future for setting boundary values or limiting numbers of work items.
+
+(2) _Preconditions_: `out_ptr` must be the same for all work-items
+in the group.
+
+._Constraints_:
+* Only available if `Group` is a work-group or sub-group.
+* `OutputIteratorT` suppose to be a random access iterator.
+* InputT must be convertible to value type of `OutputIteratorT`.
+
+_Effects_: Stores single element `in` to `out_ptr` by using the `g` group
+object to identify memory location as `out_ptr` + `g.get_local_linear_id()`
+
+`Properties` argument is reserved for future revisions of this extention and is
+ignored now. Default value is empty `sycl::ext::oneapi::experimental::properties<>`
+May be used in future for setting boundary values or limiting numbers of work items.
+
+== `sycl::vec` Functions
```

Comment:
`sycl::marray` has iterators, so it can use sycl::span API, as:

`
sycl::marray<T, items_per_thread> marray;
sycl::span<T, items_per_thread>{ marray.begin(), marray.end() };`

Do you think it needs an extra overload? (for sycl::vec it is a bit more ugly, that is why we added an extra overload)